app/src/main/java/com/example/taskwise_eventmaster/domain/repository/RoomEventRepository.kt

```diff
@@ -0,0 +1,165 @@
+package com.example.taskwise_eventmaster.domain.repository
+
+import com.example.taskwise_eventmaster.data.network.datasources.RemoteEventDataSource
+import com.example.taskwise_eventmaster.data.persistance.daos.EventDao
+import com.example.taskwise_eventmaster.data.persistance.daos.ThumbnailDao
+import com.example.taskwise_eventmaster.data.persistance.model.Thumbnail as ThumbnailPe
+import com.example.taskwise_eventmaster.domain.model.Event
+import com.example.taskwise_eventmaster.data.persistance.model.Event as EventPe
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import kotlinx.datetime.toJavaLocalDateTime
+import kotlinx.datetime.toLocalDateTime
+import javax.inject.Inject
+
+class RoomEventRepository @Inject constructor(
+    private val dataSource: RemoteEventDataSource,
+    private val dao: EventDao,
+    private val daoThumbnail: ThumbnailDao
+) : EventRepository {
+    override suspend fun saveEventLocal(event: Event) = withContext(Dispatchers.IO) {
+        try {
+
+            val eventPe = EventPe(
+                id = event.id,
+                type = event.type,
+                name = event.name,
+                dateTime_Utc = event.dateTime_Utc,
+                address = event.address,
+                city = event.city,
+                country = event.country,
+            )
+
+            dao.upsertEvent(eventPe)
+
+            for (thumbnail in event.thumbnails) {
+
+                val thumbnailPe = ThumbnailPe(
+                    thumbnailUrl = thumbnail.thumbnailUrl,
+                    eventId = event.id,
+                )
+
+                daoThumbnail.upsertThumbnail(thumbnailPe)
+            }
+        } catch (e: Exception) {
+            println("Couldn't upsert an event")
+        }
+    }
+
+    override suspend fun deleteEventLocal(eventId: Int) {
+        try {
+            daoThumbnail.deleteThumbnailsFromEvent(eventId)
+            dao.deleteEvent(eventId)
+        } catch (e: Exception) {
+            println("Error during deleting an event")
+        }
+    }
+
+    override suspend fun getEventRemote(eventId: Int): Event? = withContext(Dispatchers.IO) {
+        try {
+            val eventNe = dataSource.getEvents()?.events?.find { event -> event.id == eventId }
```

Comment:
Every time you need one event the query fetches all of them and then filters out the needed one. The better option would be to fetch only the one with the given `eventId`. Does the API provide that?